### PR TITLE
Fix: Allow Email to re-render when data changes

### PR DESCRIPTION
**Pull request by NikxDa**
This PR fixes https://github.com/silverstripe/silverstripe-framework/issues/9289

### CODE REVIEW

**Comment by NikxDa** [comment]
PHPUnit reporting 100% on the updated `EmailTest`.

File: src/Control/Email/Email.php [hunk 1]
```diff
@@ -570,6 +570,7 @@ public function getData()
     public function setData($data)
     {
         $this->data = $data;
+        $this->setBody(null);
```
Comment (maxime-rainville) [inline src/Control/Email/Email.php:573]: Might be interesting to create a `invalidateBody` method that would be an alias for `$this->setBody(null);`.

That would just make what we are doing here a bit more explicit. But that's just a nice to have ... I'm fine with the way it is right.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Looks good ... tested it locally and it works as expected. Unit tests looks thorough.

Could you target the change to the 4.7 branch so we can chip it in a 4.7 patch release?

**Comment by NikxDa** [comment]
@maxime-rainville Now targeted against the 4.7 branch

File: src/Control/Email/Email.php (hunk 1 above)
Comment (NikxDa) [reply src/Control/Email/Email.php]: Done :)

File: src/Control/Email/Email.php [hunk 2]
```diff
@@ -635,6 +640,15 @@ public function setBody($body)
         return $this;
     }
 
+    /**
+     * @return $this
+     */
+    public function invalidateBody() {
```
Comment (maxime-rainville) [inline src/Control/Email/Email.php:646]: Our linter wants this to be a separate line 

```suggestion
    public function invalidateBody() 
    {
```

**Review by maxime-rainville — APPROVED** [review]
Merge once the linting error is fix.

File: src/Control/Email/Email.php (hunk 2 above)
Comment (NikxDa) [reply src/Control/Email/Email.php]: Have added a new commit.

**Pull request merged**